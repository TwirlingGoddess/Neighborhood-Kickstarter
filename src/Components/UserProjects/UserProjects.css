.user-selected-projects-section {
  display: flex;
  height: 100%;
  width: 100%;
  /* padding: 2rem; */
}

.edit-project-section {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 100vh;
}

.selected-project {
  height: 80vh;
  width: 50%;
}

.edit-resources {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100vh;
  width: 50%;
  overflow: scroll;
}

.edit-selected-project {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  padding-top: 4rem;
  width: 50%;
}

.edit-project-info {
  height: 40rem;
  background-image: linear-gradient(to top, #c4c5c7 0%, #dcdddf 52%, #ebebeb 100%);  width: 20rem;
  padding: 2rem;
  border-radius: .7rem;
  border: 1px solid white;
}

.selected-info {
  color: white;
}

.resources {
  width: 80%;
  height: 10rem;
}

.project-photo {
  width: 20rem;
}

.edit-resource-button {
  box-sizing: border-box;
  appearance: none;
  background-color: transparent;
  border-radius: 0.6em;
  cursor: pointer;
  align-self: center;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1;
  margin: 20px;
  padding: 1em 1em;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 700;
  border-color: #3498db;
  color: #fff;
  box-shadow: 0 0 40px 40px #3498db inset, 0 0 0 0 #3498db;
  transition: all 150ms ease-in-out;
  width: 12rem;
  max-height: 4rem;
}

.edit-resource-button:hover {
  color: #fff;
  outline: 0;
  box-shadow: 0 0 10px 0 #3498db inset, 0 0 10px 4px #3498db;
}