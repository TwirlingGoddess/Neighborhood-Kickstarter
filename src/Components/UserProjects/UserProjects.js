import React, { Component } from 'react';
import { UserProjectsContainer } from '../UserProjectsContainer/UserProjectsContainer'
import { getUserProjects } from '../../utilities/apiCalls/apiCalls'

class UserProjects extends Component {
  constructor() {
    super();
    this.state = {
      userProjects: []
    }
  }

  componentDidMount = () => {
    this.getUserProjects();
  }

  getUserProjects = async () => {
    let userId = this.props.currentUser.id;

    if(userId) {
      let userProjects = await getUserProjects(userId);
  
      this.setState({
        userProjects
      })
    }
  }
  
  render() {
    return (
      <div className='user-projects-section'>
        <UserProjectsContainer userProjects={this.state.userProjects}/>
      </div>
    )
  }
}

export default UserProjects;