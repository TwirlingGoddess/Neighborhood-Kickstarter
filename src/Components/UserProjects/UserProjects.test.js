import React from 'react';
import UserProjects  from './UserProjects';
import { shallow } from 'enzyme';

describe('UserProjects', () => {
  let wrapper;
  it('should match the snapshot', () => {
    wrapper = shallow(<UserProjects />);
    expect(wrapper).toMatchSnapshot();
  });
});