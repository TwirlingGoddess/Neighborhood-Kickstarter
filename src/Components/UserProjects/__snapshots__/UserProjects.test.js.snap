// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProjects should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProjects />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <UserProjectsContainer
        selectProject={[Function]}
        userProjects={Array []}
      />,
      "className": "user-selected-projects-section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "currentUser": undefined,
        "selectProject": [Function],
        "userProjects": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <UserProjectsContainer
          selectProject={[Function]}
          userProjects={Array []}
        />,
        "className": "user-selected-projects-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentUser": undefined,
          "selectProject": [Function],
          "userProjects": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
